contracts/sol4/mock/ConversionRatesEnhancedOpen.sol

```diff
@@ -0,0 +1,106 @@
+pragma solidity 0.4.18;
+
+import "../reserves/fprConversionRate/ConversionRateEnhancedSteps.sol";
+
+/**
+ *   @title ConversionRateEnhancedOpen
+ *   Inherits ConversionRateEnhancedSteps.
+ *   Supports API to check new getRate logic by exposing internal data.
+ *   Additional API enables 2 options: 
+ *      - get rate queries that also show steps data (accumulated Y value).
+ *      - query get rate with fake imbalance.
+ */
+contract ConversionRateEnhancedOpen is ConversionRateEnhancedSteps {
+
+    function ConversionRateEnhancedOpen(address _admin) public 
+        ConversionRateEnhancedSteps(_admin)
+        { } // solhint-disable-line no-empty-blocks
+
+    ///@dev enables calling get rate and watching extra BPS from step function.
+    ///@dev doesn't check: token listed, valid rate duration.
+    ///         this logic isn't required since only used to show getRate internal values. extra logic isn't
+    ///@dev rateWithSteps value should be equal to rate value from getRate call
+    function getRateOpenData(
+        ERC20 token, 
+        uint currentBlockNumber, 
+        bool buy, 
+        uint qty 
+    ) public view returns(
+        uint rateWithSteps,
+        uint rateWithoutSteps,
+        int extraBpsYdata
+    ) {
+        // get rate update block
+        bytes32 compactData = tokenRatesCompactData[tokenData[token].compactDataArrayIndex];
+
+        uint updateRateBlock = getLast4Bytes(compactData);
+        if (currentBlockNumber >= updateRateBlock + validRateDurationInBlocks) return (0, 0, 0); // rate is expired
```

Comment:
it does check for valid rate duration here, but comment mentions it doesn't check

Reply:
will remove this check
also taking block.number instead of user input block number.
Not sure if we ever use this contract again. but better its easy to use.